Utils/azuretools-core/src/com/microsoft/azure/toolkit/lib/common/task/AzureTask.java

```diff
@@ -22,39 +22,78 @@
 
 package com.microsoft.azure.toolkit.lib.common.task;
 
-import lombok.Data;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
 
-@Data
-public class AzureTask {
-    private Modality modality;
-    private Runnable runnable;
-    private Object project;
-    private boolean cancellable;
-    private String title;
+import java.util.function.Supplier;
+
+@Getter
+public class AzureTask<T> {
+    private final Modality modality;
+    private final Supplier<T> supplier;
+    private final Object project;
+    private final boolean cancellable;
+    @Builder.Default
+    private final boolean backgroundable = true;
+    private final String title;
+
+    @Setter(AccessLevel.PACKAGE)
+    private AzureTaskContext.Node context;
 
     public AzureTask(Runnable runnable) {
-        this.runnable = runnable;
-        this.modality = Modality.DEFAULT;
+        this(runnable, Modality.DEFAULT);
```

Comment:
create a builder for `AzureTask`, as there are too many overloader for constructor which may be a bit confusing

Reply:
builder is valid but not sure if it's good here. 
since I want to emphasize that some parameters are required(e.g. the runnable/supplier).